fe/fe-core/src/main/java/com/starrocks/sql/common/PRangeCell.java

```diff
@@ -51,10 +51,11 @@ public Range<PartitionKey> getRange() {
      */
     @Override
     public int compareTo(PRangeCell o) {
-        if (isIntersected(o)) {
-            return 0;
+        int res = this.range.lowerEndpoint().compareTo(o.range.lowerEndpoint());
```

Comment:
Thank you for your PR. The implementation of the compareTo function does have issues, but it was designed that way for a specific purpose. Directly modifying it to a correct implementation would cause the partition mapping functionality to fail. We will refactor this part of the logic based on your PR.